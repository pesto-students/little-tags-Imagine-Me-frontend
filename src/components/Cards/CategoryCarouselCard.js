import { FormattedMessage } from 'react-intl'
import { Link } from 'react-router-dom'

export default function CategoryCarouselCard({ title, image }) {
    return <Link to={`/category/${title}`}>
        <div className="relative rounded-full bg-center bg-contain bg-no-repeat shadow-md w-16 h-16 mr-5 sm:rounded sm:w-48 sm:h-56 sm:bg-cover delay-200 transition-all transform hover:scale-110 my-4"
            style={{ backgroundImage: `url(${image})` }} >
            <div className="absolute bottom-5 text-center left-0 right-0 hidden sm:block">
                <FormattedMessage id={title} />
            </div>
        </div>
    </Link>
}