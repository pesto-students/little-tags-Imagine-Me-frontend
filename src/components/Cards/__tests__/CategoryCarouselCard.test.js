import Enzyme from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
Enzyme.configure({ adapter: new Adapter() })

import { IntlProvider } from 'react-intl'

const { mount } = Enzyme

import CategoryCarouselCard from '../CategoryCarouselCard'


import translation_en from '../../../translations/en.json'

describe('<CategoryCarouselCard />', () => {
    it('should renders correctly (snapshot)', () => {
        const tree = mount(
            <IntlProvider locale="en" messages={translation_en}>
                <CategoryCarouselCard
                    title="Bags"
                    image="image" />
            </IntlProvider>)
        expect(toJson(tree)).toMatchSnapshot()
    });
})