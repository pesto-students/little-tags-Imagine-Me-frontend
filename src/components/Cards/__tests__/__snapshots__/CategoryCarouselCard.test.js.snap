// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryCarouselCard /> should renders correctly (snapshot) 1`] = `
<IntlProvider
  defaultFormats={Object {}}
  defaultLocale="en"
  formats={Object {}}
  locale="en"
  messages={
    Object {
      "Bags": "Bags",
      "Hats": "Hats",
      "Jeans": "Jeans",
      "Laptop": "Laptop",
      "T-Shirt": "T-Shirt",
      "Trending": "Trending",
      "Wear Trending Collections": "Wear Trending Collections",
      "cart": "cart",
      "explore": "explore",
      "login": "login",
      "loginNow": "Login Now",
      "loginToStart": "Get started by logging in.",
      "register": "Register",
      "registerNow": "Register Now",
      "registerToStart": "Get started by Registering yourself.",
      "signInInstead": "Sign In Instead",
      "signUpInstead": "Sign Up Instead",
      "trending": "trending",
    }
  }
  onError={[Function]}
  textComponent={Symbol(react.fragment)}
>
  <CategoryCarouselCard
    image="image"
    title="Bags"
  >
    <div
      className="relative rounded-full bg-center bg-contain bg-no-repeat shadow-md w-16 h-16 mr-5 sm:rounded sm:w-48 sm:h-56 sm:bg-cover transform hover:scale-110 my-4"
      style={
        Object {
          "backgroundImage": "url(http://127.0.0.1:3001/image)",
        }
      }
    >
      <div
        className="absolute bottom-5 text-center left-0 right-0 hidden sm:block"
      >
        <FormattedMessage
          id="Bags"
        >
          Bags
        </FormattedMessage>
      </div>
    </div>
  </CategoryCarouselCard>
</IntlProvider>
`;
